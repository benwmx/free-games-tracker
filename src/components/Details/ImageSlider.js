/* eslint-disable react/require-default-props */
/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { Slide } from 'react-slideshow-image';
import 'react-slideshow-image/dist/styles.css';

const ImageSlider = (props) => {
  const { screenshots, thumbnail } = props;

  return (
    <div className="imageSlider">
      {
        (screenshots.length > 0)

          ? (
            <Slide easing="ease">
              {
          screenshots.map((screenshot) => (
            <img key={screenshot.id} src={screenshot.image} style={{ width: '100%', backgroundSize: 'content' }} alt="game screenshot not available" />
          ))
        }
            </Slide>
          )
          : <img src={thumbnail} alt="A Games thumbnail" />
      }
    </div>
  );
};

ImageSlider.propTypes = {
  screenshots: PropTypes.array,
  thumbnail: PropTypes.string,
};

// {
//   (screenshots.length > 0)
//     ? screenshots.map((screenshot) =>
//     (<img key={screenshot.id} src={screenshot.image} alt="A Games thumbnail" />))
//     : <img src={thumbnail} alt="A Games thumbnail" />
// }

export default ImageSlider;
