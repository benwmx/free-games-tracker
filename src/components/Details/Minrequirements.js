import React from 'react';

function Minrequirements() {
  return (
    <div className="minReqsContainer">
      <p>Minimum System Requirements</p>
      <table className="minReqsTable">
        <thead />
        <tbody>
          <tr>
            <td>OS</td>
            <td>Bobo</td>
          </tr>
          <tr>
            <td>processor</td>
            <td>Bobo</td>
          </tr>
          <tr>
            <td>Memory</td>
            <td>Bobo</td>
          </tr>
          <tr>
            <td>Graphics</td>
            <td>Bobo</td>
          </tr>
          <tr>
            <td>Storage</td>
            <td>Bobo</td>
          </tr>
        </tbody>
      </table>
    </div>
  );
}

export default Minrequirements;
