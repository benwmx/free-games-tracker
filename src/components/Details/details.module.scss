@import url('https://fonts.googleapis.com/css2?family=Patrick+Hand&display=swap');

.container {
  min-height: 100vh;
  margin-top: 50px;
  font-family: 'Patrick Hand', cursive;
}

.containerDark {
  @extend .container;

  background-image: url('../../img/webb-dark.png');
  color: white;
}

.containerLight {
  @extend .container;

  background-image: url('../../img/webb.png');
}

.screenShotsContainer {
  min-height: 100px;
}

a {
  text-decoration: underline;
  color: black;
}

.detailsContainer {
  display: flex;
  flex-flow: column;
  background-color: rgba(0, 0, 0, 0.1);

  .title {
    background-color: red;
    margin: 0;
    text-align: center;
    padding: 10px;
    translate: 0 -5px;
  }

  .genre,
  .releaseDate,
  .developer,
  .publisher,
  .link {
    display: flex;
    align-items: center;
    border-bottom: 2px solid red;
    padding: 0 20px;

    .name {
      font-size: 1.3rem;
      flex: 0.7;
    }

    .value {
      flex: 1.3;
      color: #4e0058;
      text-align: center;
      background-color: rgb(248, 159, 159);
      padding: 2px;
      border-radius: 6px;
      -webkit-border-radius: 6px;
      -moz-border-radius: 6px;
      -ms-border-radius: 6px;
      -o-border-radius: 6px;
    }
  }

  .descContainer {
    display: flex;
    flex-flow: column;
    text-align: center;
    border-bottom: 2px solid red;

    .descName {
      font-size: 1.3rem;
      margin: 20px 0 0 0;
    }

    .description {
      padding: 0 20px;
    }
  }

  .minReqs {
    border-bottom: 2px solid red;

    .minReqsContainer {
      margin: 0 10px 20px 10px;

      .name {
        font-size: 1.3rem;
        text-align: center;
      }

      .minReqsTable {
        margin: 0 auto;
        border: 2px solid red;
        border-collapse: collapse;

        td {
          border: 1px solid red;
          text-align: center;
          font-size: 1.2rem;
          padding: 6px;
        }
      }
    }
  }
}

@media screen and (min-width: 700px) {
  .detailsContainer {

    .genre,
    .releaseDate,
    .developer,
    .publisher,
    .link {
      
      .name {
        flex: 1.3;
        font-size: 1.5rem;
      }

      .value {
        flex: 0.7;
        font-size: 1.4rem;
      }
    }
    ;
  }
}
