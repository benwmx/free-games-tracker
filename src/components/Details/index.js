import React from 'react';
import { useSelector } from 'react-redux';
import ImageSlider from './ImageSlider';
import Minrequirements from './Minrequirements';
import Styles from './details.module.scss';

const Details = () => {
  const details = useSelector((state) => state.details);
  const theme = useSelector((state) => state.theme);

  return (
    <div className={(theme.darkTheme) ? Styles.containerDark : Styles.containerLight}>
      <div className={Styles.screenShotsContainer}>
        {
        (details.screenshots === undefined)
          ? 'Loading ...'
          : <ImageSlider screenshots={details.screenshots} thumbnail={details.thumbnail} />
      }
      </div>
      <div className={Styles.detailsContainer}>
        <h1 className={Styles.title}>{details.title}</h1>
        <div className={Styles.genre}>
          <p className={Styles.name}>Genre : </p>
          <p className={Styles.value}>{details.genre}</p>
        </div>
        <div className={Styles.releaseDate}>
          <p className={Styles.name}>Release Date : </p>
          <p className={Styles.value}>{details.release_date}</p>
        </div>
        <div className={Styles.developer}>
          <p className={Styles.name}>Developer : </p>
          <p className={Styles.value}>{details.developer}</p>
        </div>
        <div className={Styles.publisher}>
          <p className={Styles.name}>Publisher : </p>
          <p className={Styles.value}>{details.publisher}</p>
        </div>
        <div className={Styles.descContainer}>
          <p className={Styles.descName}>Description : </p>
          <p className={Styles.description}>
            {details.description}
          </p>
        </div>
        <div className={Styles.minReqs}>
          {
            (details.minimum_system_requirements !== undefined)
              ? <Minrequirements requirements={details.minimum_system_requirements} />
              : ''
          }
        </div>
        <div className={Styles.link}>
          <p className={Styles.name}>Link : </p>
          <p className={Styles.value}>
            {
              (details.game_url !== undefined)
                ? <a href={details.game_url}>ğŸ”— more on official Website ğŸ”— </a>
                : 'No Link Available'
            }
          </p>
        </div>
      </div>
    </div>
  );
};
export default Details;
