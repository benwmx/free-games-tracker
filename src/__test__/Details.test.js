import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Details from '../components/Details';

describe('Details Component test', () => {
  it('Should render', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Router>
          <Details />
        </Router>
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
