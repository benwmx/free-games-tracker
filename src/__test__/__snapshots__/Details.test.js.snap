// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component test Should render 1`] = `
<div
  className="containerLight"
>
  <div
    className="screenShotsContainer"
  >
    Loading ...
  </div>
  <div
    className="detailsContainer"
  >
    <h1
      className="title"
    />
    <div
      className="genre"
    >
      <p
        className="name"
      >
        Genre : 
      </p>
      <p
        className="value"
      />
    </div>
    <div
      className="releaseDate"
    >
      <p
        className="name"
      >
        Release Date : 
      </p>
      <p
        className="value"
      />
    </div>
    <div
      className="developer"
    >
      <p
        className="name"
      >
        Developer : 
      </p>
      <p
        className="value"
      />
    </div>
    <div
      className="publisher"
    >
      <p
        className="name"
      >
        Publisher : 
      </p>
      <p
        className="value"
      />
    </div>
    <div
      className="descContainer"
    >
      <p
        className="descName"
      >
        Description : 
      </p>
      <p
        className="description"
      />
    </div>
    <div
      className="minReqs"
    >
      
    </div>
    <div
      className="link"
    >
      <p
        className="name"
      >
        Link : 
      </p>
      <p
        className="value"
      >
        No Link Available
      </p>
    </div>
  </div>
</div>
`;
